oidc-wp/openid-connect-generic
# Fixes Broken Redirect URL Handling & Moves Away from Cookies

### All Submissions:






### Changes proposed in this Pull Request:



- Fixes #233 .
- Fixes #207 .
- Fixes #57 .
- Partial fix for #285 .
- Partial fix for #159 .
- Add Redirection via State Transient Support.

### How to test the changes in this Pull Request:

1. Add the plugin to a site.
2. Activate & configure plugin.
3. Load up the WordPress login screen or add Login button shortcode to a page.
4. Observe the redirect cookie is no longer being set.
5. Check site transients and observe that the State transients are now arrays.
6. Observe the redirect_to values match expected behavior.

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
- Fixes problems with Redirect User Back plugin setting no longer working.
- Fixes issues where redirect cookies fail to be set and rely on state.
- Fixes shortcodes to properly handle attributes.
